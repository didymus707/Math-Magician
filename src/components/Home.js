import React from 'react';
import Nav from './Nav';

const Home = () => (
  <>
    <Nav />
    <main>
      <div className="contents">
        <h2>Welcome to our Page!</h2>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit.
          Maiores reprehenderit animi, quisquam voluptatum magnam corporis dicta dolorem tenetur
          enim repellendus dolorum possimus facilis sed illo laudantium, esse iste aliquid porro!
        </p>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit.
          Maiores reprehenderit animi, quisquam voluptatum magnam corporis dicta dolorem tenetur
          enim repellendus dolorum possimus facilis sed illo laudantium, esse iste aliquid porro!
        </p>
      </div>
    </main>
  </>
);

export default Home;
