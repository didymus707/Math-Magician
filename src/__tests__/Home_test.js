import React from 'react';
import renderer from 'react-test-renderer';
import TestUtils from 'react-dom/test-utils';
import { BrowserRouter, Route } from 'react-router-dom';
import Home from '../components/Home';

describe('Home', () => {
  it('wraps content within a main tag', () => {
    const main = TestUtils.renderIntoDocument(
      <BrowserRouter>
        <Route exact path="/" component={Home} />
      </BrowserRouter>,
    );
    TestUtils.findRenderedDOMComponentWithTag(
      main,
      'main',
    );
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Route exact path="/" component={Home} />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
