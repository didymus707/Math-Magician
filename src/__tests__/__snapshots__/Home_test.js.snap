// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
Array [
  <nav
    className="nav-con"
  >
    <h1>
      Math Magician
    </h1>
    <ul>
      <li
        className="home"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="calc"
      >
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calulator
        </a>
      </li>
      <li
        className="quo"
      >
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>,
  <main>
    <div
      className="contents"
    >
      <h2>
        Welcome to our Page!
      </h2>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Maiores reprehenderit animi, quisquam voluptatum magnam corporis dicta dolorem tenetur enim repellendus dolorum possimus facilis sed illo laudantium, esse iste aliquid porro!
      </p>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Maiores reprehenderit animi, quisquam voluptatum magnam corporis dicta dolorem tenetur enim repellendus dolorum possimus facilis sed illo laudantium, esse iste aliquid porro!
      </p>
    </div>
  </main>,
]
`;
